spinalcordtoolbox/scripts/sct_dmri_denoise_patch2self.py

```diff
@@ -0,0 +1,174 @@
+    )
+
+    mandatory = parser.add_argument_group("\nMANDATORY ARGUMENTS")
+    mandatory.add_argument(
+        "-i",
+        default=None,
+        required=True,
+        help="Input NIfTI image to be denoised. Example: image_input.nii.gz",
+        metavar=Metavar.file,
+    )
+    mandatory.add_argument(
+        "-b",
+        default=None,
+        required=True,
+        help="Input bvals file corresponding to the NIfTI file to be denoised."
+             " Example: filename.bval",
+        metavar=Metavar.file,
+    )
+
+    optional = parser.add_argument_group("\nOPTIONAL ARGUMENTS")
+    optional.add_argument(
+        "-h",
+        "--help",
+        action="help",
+        help="Show this help message and exit")
+    optional.add_argument(
+        "-model",
+        help='Type of regression model used for self-supervised training within Patch2Self.',
+        required=False,
+        choices=("ols", "ridge", "lasso"),
+        default="ols")
+    optional.add_argument(
+        "-radius",
+        help="Patch Radius used to generate p-neighbourhoods within Patch2Self. Notes:\n"
+             "- A radius of '0' will use 1x1x1 p-neighbourhoods, a radius of '1' will use "
+             "3x3x3 p-neighbourhoods, and so on.\n"
+             "- For ansiotropic patch sizes, provide a comma-delimited list of 3 integers. "
+             "(e.g. '-radius 0,1,0'). For isotropic patch sizes, provide a single int value "
+             "(e.g. '-radius 0').",
+        metavar=Metavar.int,
+        required=False,
+        default="0")
+    optional.add_argument(
+        "-o",
+        help="Name of the output NIFTI image.",
+        metavar=Metavar.str,
+        default=None)
+    optional.add_argument(
+        "-r",
+        help="Remove temporary files. Specify 0 to get access to temporary files.",
+        type=int,
+        choices=(0, 1),
+        default=1)
+    optional.add_argument(
+        '-v',
+        metavar=Metavar.int,
+        type=int,
+        choices=[0, 1, 2],
+        default=1,
+        # Values [0, 1, 2] map to logging levels [WARNING, INFO, DEBUG], but are also used as "if verbose == #" in API
+        help="Verbosity. 0: Display only errors/warnings, 1: Errors/warnings + info messages, 2: Debug mode")
+
+    return parser
+
+
+def main(argv=None):
+    parser = get_parser()
+    arguments = parser.parse_args(argv)
+    verbose = arguments.v
+    set_loglevel(verbose=verbose)
+
+    model = arguments.model
+    remove_temp_files = arguments.r
+    if "," in arguments.radius:
+        patch_radius = list_type(",", int)(arguments.radius)
+    else:
+        patch_radius = int(arguments.radius)
+
+    file_to_denoise = arguments.i
+    bval_file = arguments.b
+    output_file_name = arguments.o
+
+    param = Param()
+    param.verbose = verbose
+    param.remove_temp_files = remove_temp_files
+    param.parameter = model
```

Comment:
Just reviewing this PR one last time with fresh eyes in the new year. :)

I think that `Param` was something that was copied from our other denoising script, but, looking at its usage, I don't think it's actually necessary here? 

I think we could simplify things by getting rid of `Param` entirely, and sticking only with `arguments` for containing the default arguments.

Reply:
Done in c6c1dd0.